pkg/iommu/iommu.go

```diff
@@ -0,0 +1,59 @@
+package iommu
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+func GroupMapForPCIDevices(groupPaths []string) map[string]int {
+	groupMap := make(map[string]int)
+	for _, groupPath := range groupPaths {
+		address := strings.Split(groupPath, "/")[6]
+		group := GroupForPCIDevice(address, groupPaths)
```

Comment:
The nested for loop can be optimized by scaning groupPaths once and save the information into a map